tests/integration/_experimental/dataset/test_evaluation.py

```diff
@@ -55,188 +52,169 @@ def get_df_dp(n: int = 20) -> pd.DataFrame:
-
-    return infer_func
-
-
-def get_eval_func(
-    df_mtr: pd.DataFrame,
-    columns: List[str],
-    id_col: str = "user_dp_id",
-    how: str = "left",
-    keep_none: bool = False,
-) -> EVAL_FUNC_TYPE:
-    def eval_func(
-        datapoints: pd.DataFrame,
-        inferences: pd.DataFrame,
-        eval_config: TYPE_EVALUATION_CONFIG,
-    ) -> pd.DataFrame:
-        _metrics = datapoints.set_index(id_col).join(df_mtr.set_index(id_col), how=how).reset_index()[columns]
-        if keep_none:
-            _metrics = _metrics.replace({np.nan: None})
-        return _metrics
-
-    return eval_func
-
-
 def test__test() -> None:
     dataset_name = with_test_prefix(f"{__file__}::test__test")
     model_name = with_test_prefix(f"{__file__}::test__test")
     df_dp = get_df_dp()
     dp_columns = ["user_dp_id", "locator", "width", "height", "city"]
     register_dataset(dataset_name, df_dp[3:10][dp_columns])
 
-    df_inf = get_df_inf()
-    df_mtr = get_df_mtr()
-    inf_columns = ["softmax_bitmap"]
-    mtr_columns = ["score"]
+    df_result = get_df_result()
+    result_columns = ["softmax_bitmap", "score"]
 
-    eval_configs = [
-        {"threshold": 0.3},
-        {"threshold": 0.6},
-    ]
     test(
         dataset_name,
         model_name,
-        infer=get_infer_func(df_inf, inf_columns),
-        eval=get_eval_func(df_mtr, mtr_columns),
-        eval_configs=eval_configs,
+        df_result,
+        on="user_dp_id",
     )
 
-    df_by_eval = fetch_evaluation_results(dataset_name, model_name)
-    assert len(df_by_eval) == 2
-    assert sorted([cfg for cfg, *_ in df_by_eval], key=lambda x: x.get("threshold")) == sorted(
-        eval_configs,
-        key=lambda x: x.get("threshold"),
-    )
+    fetched_df_dp, df_results_by_eval = fetch_results(dataset_name, model_name)
+    eval_cfg, fetched_df_result = df_results_by_eval[0]
+    assert len(df_results_by_eval) == 1
+    assert eval_cfg is None
     expected_df_dp = df_dp[3:10].reset_index(drop=True)
-    expected_df_inf = df_inf[3:10].reset_index(drop=True)
-    expected_df_mtr = df_mtr[3:10].reset_index(drop=True)
-    for df_eval in df_by_eval:
-        _assert_frame_equal(df_eval[1], expected_df_dp, dp_columns)
-        _assert_frame_equal(df_eval[2], expected_df_inf, inf_columns)
-        _assert_frame_equal(df_eval[3], expected_df_mtr, mtr_columns)
+    expected_df_result = df_result[3:10].reset_index(drop=True)
+    _assert_frame_equal(fetched_df_dp, expected_df_dp, dp_columns)
+    _assert_frame_equal(fetched_df_result, expected_df_result, result_columns)
 
 
-def test__test__missing_inference() -> None:
-    dataset_name = with_test_prefix(f"{__file__}::test__test__missing_inference")
-    model_name = with_test_prefix(f"{__file__}::test__test__missing_inference")
+def test__test__align_manually() -> None:
+    dataset_name = with_test_prefix(f"{__file__}::test__test__align_manually")
+    model_name = with_test_prefix(f"{__file__}::test__test__align_manually")
```

Comment:
don't think this use case is simpler for end users, we might need to require `on` column all the time, but leave it as this for now based on the current discussion